tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,173 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	runningTime := 60 * time.Second // time for which operation should be performed
+	noOfPrefixes := 20              // total number of prefixes which will be watched upon
+	watchPerPrefix := 10            // number of watchers per prefix
+	reqRate := 20                   // put request per second
+	keyPrePrefix := 1000            // max number of keyPrePrefixs for put operation
+
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	prePrefixKeys := generateRandomKeys(10, keyPrePrefix)
+	// map for holding prefixes and values
+	prefixKeysMap := make(map[string][]string)
+	// actual keys for performing put operation
+	keys := make([]string, 0)
+
+	// initialize the map with all prefixes
+	for _, prefix := range prefixes {
```

Comment:
We defined number of prefix and number of keys per prefix, and number of watcher per prefix around line 223.

each prefix should have exact same number of keys as we defined.
each watcher should receive the exact number of events as we defined.

Also as I mentioned, i feel what you suggested is not necessary for now.